src/Http/Middlewares/DeprecatedRoute.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Jobilla\DeprecatedRoutes\Http\Middlewares;
+
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Event;
+use Illuminate\Support\Facades\Log;
+use Jobilla\DeprecatedRoutes\Events\RouteIsDeprecated;
+
+class DeprecatedRoute
+{
+    /**
+     * @var string
+     */
+    public const HEADER_NAME = 'X-Deprecated-At';
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  mixed  $deprecatedAt
+     * @return mixed
+     */
+    public function handle(Request $request, \Closure $next, $deprecatedAt): Response
+    {
+        /** @var \Illuminate\Http\Response $response */
+        $response = $next($request);
+
+        $this->logDeprecatedRequest($request);
+        $this->fireDeprecationEvent($request, $response);
+
+        $response->header(static::HEADER_NAME, $this->getTimestampString($deprecatedAt));
+
+        return $response;
+    }
+
+    protected function logDeprecatedRequest(Request $request)
+    {
+        $level = Config::get('deprecated_routes.log_level');
+
+        if ($level === false) {
+            return;
+        }
+
+        Log::log($level, 'Deprecated route on ' . $request->path() . ' was requested.');
+    }
+
+    protected function fireDeprecationEvent(Request $request, Response $response)
+    {
+        $enabled = Config::get('deprecated_routes.fire_event');
+
+        if ($enabled !== true) {
```

Comment:
Why is this one `!== true` and the other one `=== false` ?

Reply:
I guess I can check for truthiness here via just a `if ($enabled)` since there is no other choices than true/false.

I checked for explicit `false` value because any other value (including empty strings and maybe zeros in the future) might be used as a logging level since Laravel apps can use any logging handler and are not limited to Monolog.

Reply:
Makes sense